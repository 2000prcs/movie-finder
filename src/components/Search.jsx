import React, { Component } from 'react';
import { Input } from 'semantic-ui-react';


export default class Search extends React.Component {
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  render() {
    return (
      <div className="search">
        <Input onChange={this.props.handleInputChange} action={{ icon: 'search' }} placeholder="Search..." />
      </div>
    );
  }
}
