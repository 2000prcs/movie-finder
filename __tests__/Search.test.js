import React from 'react';
import Search from '../src/components/Search.jsx';


describe('Search component', () => {
  test('Should render Search', () => {
    const SearchWrapper = shallow(<Search />);
    expect(SearchWrapper).toMatchSnapshot();
  });
});
