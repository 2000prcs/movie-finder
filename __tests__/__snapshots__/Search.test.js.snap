// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component Should render Search 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Input
          icon={true}
          placeholder="Type Movie Title Here!"
          type="text"
        >
          <input />
          <Icon
            as="i"
            circular={true}
            name="search"
          />
        </Input>,
        <Label
          pointing="left"
        >
          Now Showing: 
          <b>
            TMDB Popular Movies
          </b>
        </Label>,
      ],
      "className": "search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <input />,
            <Icon
              as="i"
              circular={true}
              name="search"
            />,
          ],
          "icon": true,
          "onChange": undefined,
          "placeholder": "Type Movie Title Here!",
          "type": "text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "i",
              "circular": true,
              "name": "search",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Now Showing: ",
            <b>
              TMDB Popular Movies
            </b>,
          ],
          "pointing": "left",
        },
        "ref": null,
        "rendered": Array [
          "Now Showing: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "TMDB Popular Movies",
            },
            "ref": null,
            "rendered": "TMDB Popular Movies",
            "type": "b",
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Input
            icon={true}
            placeholder="Type Movie Title Here!"
            type="text"
          >
            <input />
            <Icon
              as="i"
              circular={true}
              name="search"
            />
          </Input>,
          <Label
            pointing="left"
          >
            Now Showing: 
            <b>
              TMDB Popular Movies
            </b>
          </Label>,
        ],
        "className": "search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <input />,
              <Icon
                as="i"
                circular={true}
                name="search"
              />,
            ],
            "icon": true,
            "onChange": undefined,
            "placeholder": "Type Movie Title Here!",
            "type": "text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "i",
                "circular": true,
                "name": "search",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Now Showing: ",
              <b>
                TMDB Popular Movies
              </b>,
            ],
            "pointing": "left",
          },
          "ref": null,
          "rendered": Array [
            "Now Showing: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "TMDB Popular Movies",
              },
              "ref": null,
              "rendered": "TMDB Popular Movies",
              "type": "b",
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
